packages/gatsby/src/utils/start-server.ts

```diff
@@ -259,6 +259,14 @@ module.exports = {
 
     if (enableRefresh && authorizedRefresh) {
       refresh(req, pluginName)
+      res.status(200)
+      res.setHeader(`content-type`, `application/json`)
+    } else {
+      res.status(authorizedRefresh ? 404 : 403)
+      res.json({
+        error: `Refresh failed`,
```

Comment:
If you want to make it easier to adopt - I'd suggest expanding on error message (stuff below will likely need to use prettier if applied):
```suggestion
        error: enableRefresh ? `Authorization failed. Make sure you add authorization header to your refresh requests` : `Refresh endpoint is not enabled. Run gatsby with ENABLE_GATSBY_REFRESH_ENDPOINT=true environtment variable set.`,
```